generator/src/main/java/com/scottlogic/deg/generator/generation/VelocityMonitor.java

```diff
@@ -0,0 +1,91 @@
+package com.scottlogic.deg.generator.generation;
+
+import com.scottlogic.deg.generator.Field;
+import com.scottlogic.deg.generator.fieldspecs.FieldSpec;
+import com.scottlogic.deg.generator.fieldspecs.RowSpec;
+import com.scottlogic.deg.generator.outputs.GeneratedObject;
+import com.scottlogic.deg.generator.walker.reductive.FixedField;
+import com.scottlogic.deg.generator.walker.reductive.ReductiveState;
+
+import java.math.BigInteger;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+
+public class VelocityMonitor implements ReductiveDataGeneratorMonitor {
+    private String startedGenerating;
+    private long rowsSinceLastSample;
+    private BigInteger rowsEmitted;
+    private BigInteger maxRows;
+    private Timer timer = new Timer(true);
+    private SimpleDateFormat simpleDateFormat = new SimpleDateFormat("hh:mm:ss");
+    private long previousVelocity = 0;
+
+    @Override
+    public void generationStarting(GenerationConfig generationConfig) {
+        this.startedGenerating = simpleDateFormat.format( new Date());
+        this.rowsSinceLastSample = 0;
+        this.rowsEmitted = BigInteger.ZERO;
+        this.maxRows = BigInteger.valueOf(generationConfig.getMaxRows());
+
+        System.out.println("Generation started at: " + this.startedGenerating + "\n");
```

Comment:
Question: the `\n` at the end of this string will present an empty line before "Number of rows..." is emitted, is this what you wanted?

Reply:
Yes, we intended to leave a blank line before the table of rows/velocity for visual clarity